pkgs/by-name/xr/xrsh/package.nix

```
@@ -0,0 +1,62 @@
+{
+  stdenv,
+  fetchFromGitea,
+  makeWrapper,
+  writeShellApplication,
+  python3Packages,
+  lib,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "xrsh";
+  version = "0.143";
+
+  src = fetchFromGitea {
+    fetchSubmodules = true;
+    domain = "codeberg.org";
+    owner = "xrsh";
+    repo = "xrsh";
+    url = "https://codeberg.org/xrsh/xrsh.git";
+    rev = "435ac2fc51d5966a86607bb5ede3661a210157d4";
+    hash = "sha256-g+YxeExomlWHF7suzlPjUlZe6eXa9MPA10bCoifQO1I=";
+  };
+
+  dontBuild = true;
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  installPhase =
+    let
+      launcher = writeShellApplication {
+        name = "xrsh-launcher";
+        runtimeInputs = [ python3Packages.twisted ];
+        text = ''
+          XRSH_ROOT="''${1}"
+          XRSH_PORT="''${XRSH_PORT-8080}"
+
+          exec twistd -n web --listen "tcp:''${XRSH_PORT}" --path "''${XRSH_ROOT}"
+        '';
+      };
+    in
+    ''
+      runHook preInstall
+
+      mkdir -p $out/{bin,share/xrsh}
+      cp -r -t $out/share/xrsh index.html src xrsh.ico xrsh.svg
```

I think you also need to copy `xrsh.js` since its symlink is missing:

```suggestion
      cp -r -t $out/share/xrsh index.html src xrsh.js xrsh.ico xrsh.svg
```

That said, there seem to be another symlink missing for `aframe`:
```
       > ERROR: noBrokenSymlinks: the symlink /nix/store/682sjjqj55rhb97lzkj5y46cxvlqswvi-xrsh-0.143/share/xrsh/src/aframe points to a missing target: /nix/store/682sjjqj55rhb97lzkj5y46cxvlqswvi-xrsh-0.143/aframe
       > ERROR: noBrokenSymlinks: found 1 dangling symlinks and 0 reflexive symlinks
```

Does `aframe` refer to something outside of the repo?

```
$ readlink src/aframe 
../../../aframe
```

---

We can ignore it,  (its not used in any way) it was a symlink for local development.
I'll add it to .gitignore, so in the next update it'll be gone

---

You pushed to the new commit that dropped the file, but also added the file to the list of files to be copied here. Please drop it again.